src/osd/PG.cc

```diff
@@ -6724,6 +6724,36 @@ void PG::_delete_some(ObjectStore::Transaction *t)
 {
   dout(10) << __func__ << dendl;
 
+  {
+    float osd_delete_sleep = osd->osd->get_osd_delete_sleep();
+    if (osd_delete_sleep > 0 && delete_needs_sleep) {
+      epoch_t e = get_osdmap()->get_epoch();
+      //PGRef pgref(this);
+      auto delete_requeue_callback = new FunctionContext([this, e](int r) {
```

Comment:
Do I need a _PGRef_ here instead of _this_ for the callback?

Reply:
yes, iirc that's why it was used in the other sleep code - if the pg gets deleted and then the callback fires, it will crash

Reply:
even though this is the pg deletion code, it may still be needed here to handle osd shutdown